src/applications/discharge-wizard/utilities/display-logic-questions.js

```diff
@@ -29,8 +30,39 @@ export const displayConditionsMet = (SHORT_NAME, formResponses) => {
   const displayConditionsForShortName = DISPLAY_CONDITIONS[SHORT_NAME];
   const questionRequirements = Object.keys(displayConditionsForShortName);
 
+  if (questionRequirements.includes('FORK')) {
```

Comment:
Given that `anyForkConditionsMet` and `allConditionsMet` are not changed a lot throughout this function, is it possible to simplify like this?

```
if (questionRequirements.includes('FORK')) {
  const forkedReqs = displayConditionsForShortName?.FORK;

  // Iterates through the entries available for the current questions reqs.
  for (const forkIndex of Object.keys(forkedReqs)) {
      const forkConditions = forkedReqs[forkIndex];

      // Iterates through the responses required and checks for valid conditions.
      for (const conditionKey of Object.keys(forkConditions)) {
          const requiredResponses = forkConditions[conditionKey];
          const formResponse = formResponses?.[conditionKey];

          if (
              !responseMatchesRequired(requiredResponses, formResponse) &&
              requiredResponses.length
          ) {
              return false;
          }
      }   
  }

  return true;
}
  ```

Reply:
Thanks! I had the logic like this before but wasn't working. Maybe I had a bug somewhere, this is working now though. Any thoughts on optimization of this logic block? I don't like the nested loop here but given the need for forked DCs I could only think of this.

Reply:
I'd fall back on using as much of the PACT Act logic as possible. It seems like the display conditions' structure for DUW are similar to those in PACT Act in complexity. Can we re-use those utilities to avoid having to re-engineer (and retest) these utilities?

Reply:
Hmm I initially tried to use the short and long fork logic but for DUW there are multiple different ways to skip ahead to different questions based on answers. IE: Prior Service will have 5-6 DCs under the FORK obj. 

Reply:
Oh ok 👍🏻 I think a nested loop is fine, in this case. We aren't dealing with hundreds or thousands of records to loop through, in this case. Performance impact should be minimal, if at all.